cartographer/mapping/map_builder.cc

```diff
@@ -177,8 +177,24 @@ void MapBuilder::LoadMap(io::ProtoStreamReader* const reader) {
   proto::SparsePoseGraph pose_graph;
   CHECK(reader->ReadProto(&pose_graph));
 
-  const int map_trajectory_id = AddTrajectoryForDeserialization();
-  sparse_pose_graph_->FreezeTrajectory(map_trajectory_id);
+  std::map<int, int> trajectory_remapping;
+  for (auto& trajectory_proto : *pose_graph.mutable_trajectory()) {
+    const int new_trajectory_id = AddTrajectoryForDeserialization();
+    // Also takes care of checking for duplicates.
```

Comment:
Prefer code over a comment, i.e. add to the `CHECK`:
```
<< "Duplicate trajectory ID: " << trajectory_proto.trajectory_id();
```
?

Reply:
Done